### PR TITLE
fixes my shit

### LINKED ISSUE #6991: Open to Abuse - Raging Mages
There's nothing stopping wizards from attacking each other while in the den. A wizard can easily spawn camp the latter wizards and take all their spell books for themselves.

This is way too open for abuse, a hole that's too large to be covered by admins alone. I'm calling for a code solution to this as well.

### LINKED ISSUE #7154: "Life" and "Strange Reagent" don't work as intended.
With the Strange Reagent, the code says its suppose to bring a person back to life if their brute/burn is below 80 and turn them into the zombie mutant race. As of right now it does all of those things expect bring them back to life. They do the body twitching and gasp before immediately dying right afterwards.

With the Life chemical, once the heat reaction goes off, your left with the chemical itself, which is useless as of right now.

### LINKED ISSUE #7269: Issues with invisibility power
- you don't turn visible when someone drags you
- you don't turn visible if someone swaps spots with you
- you don't turn visible if pushed down or disarmed
- you don't turn visible when bumped around on non-help intent

can we start testing things before merging them

### LINKED ISSUE #7297: Liquid Dark Matter no bueno
It just does NOTHING

### LINKED ISSUE #7497: Balance request: simplify Oculine's recipe
Has anyone tried to mix oculine? It's bad. Just bad. This is the sort of recipe I'd expect for a secret goonchem, not something you can get from grinding carrots.

Oculine requires atropine, spaceacillin, and saline-glucose solution.

Atropine takes five different chemicals, of which three of them are themselves mixed (phenol, acetone, diethylamine), of which two contain oil which must itself be mixed (phenol and acetone).

Spaceacillin takes two chemicals. Epinephrine itself is a chore to mix, again requiring phenol, acetone and diethylamine. It also takes cryptobiolin, another mixed drug. On a side note, tgstation wiki has spaceacillin's recipe wrong, but that's a wiki problem.

Saline-glucose solution is easy enough to make in large amounts, but it's just yet another mixed drug on top of the others.

The recipe is laughably complex and cumbersome. Using just 5u of the basic chems whenever possible, you still have to repeatedly filter excess chems out of your beaker just to make the atropine and spaceacillin fit in the beaker. It's a lot of moving the beaker back and forth between different machines, organizing all the different chems involved.

All this effort will produce a very large supply of eye medication. It's large because you have to use so many chemicals to make it. How many people come by asking for eye meds? Not nearly enough to be worth it. You'll end the round with 90% of the medicine wasted, almost every time. I suppose you could mix only part of it and use the spare atropine to... I don't know. Nobody uses atropine. The odds are low of a med doctor even knowing what atropine is for in the first place, and it's almost always better to use the (several times buffed) basic standby chems like salglu, styptic, and silversulf. Mix in some perfluorodecalin and synthflesh if you want a super get-out-of-crit-fast patch. If they're near medbay, put them in cryo. Leftover spaceacillin and cryptobiolin are similarly unlikely to serve any purpose in most rounds. Crypto in particular is otherwise only useful for antags, and there are a dozen better options for that.

For comparison, ear medicine is simply charcoal+carbon+water. Charcoal kind of takes time, but it doesn't involve juggling beakers full of six different chemicals for two of its three components. You also have a bunch of charcoal easily available from round start. Brain damage is even easier to fix, which is a good thing because trying to make oculine will fry your head. The pre-goonchem recipe for eye damage was piss easy, as well.

### PR DESCRIPTION
**Pull request by Iamgoofball**
Fixes #7297
Fixes #7269
Fixes #7154
Fixes #6991
Fixes #7497

### CODE REVIEW

**Comment by Pennwick** [comment]
Won't this make space drugs grenades impossible? Setting the temp of every grenade to 1000 seems lazy. At the very least couldn't the grenade take the temprature of the first beaker put into it instead?

**Comment by Iamgoofball** [comment]
my "lore" is that the igniter superheats the contents of the beakers when combining them

but yeah i'll look into it

**Comment by TZK13** [comment]
Could always just change the recipe for neurotoxin.

**Comment by Pennwick** [comment]
Thats an exeptionally lazy and shortsighted excuse. There are instances where you don't want a heat reaction to take place in your grenade and there may be more in the future.  Nor does it make much sense as most grenades used are just a wire wrapped around a folded sheet of metal, no ignighter included.

**Comment by TZK13** [comment]
That's true, most grenades that chemists make don't include igniters.

File: code/game/gamemodes/wizard/spellbook.dm [hunk 1]
```diff
@@ -28,6 +29,13 @@
 
 
 /obj/item/weapon/spellbook/attack_self(mob/user as mob)
+	if(!owner)
+		user << "You bind the spellbook to yourself."
+		owner = user
+		return
+	if(user != owner)
+		user << "You can't open the spellbook somehow."
```
Comment (GunHog) [inline code/game/gamemodes/wizard/spellbook.dm]: How about something more nerdy? Such as "a mysterious force prevents you opening the [name]" or something.

Comment (Deantwo) [inline code/game/gamemodes/wizard/spellbook.dm]: Would be better if it actaully explain why you can't open it too.
Like: ```This [name] does now recognize you as its owner and refuse to open.```

**Comment by Iamgoofball** [comment]
@GunHog what

i thought you need igniter+X TRIGGER METHOD to make a 'nade

**Comment by Deantwo** [comment]
When did you last make a grenade? Over a year ago?
Read this: https://tgstation13.org/wiki/Grenade

**Comment by Iamgoofball** [comment]
this shows my knowledge of game mechanics is deep

yeah i'll look into it

**Comment by Iamgoofball** [comment]
okay @TZK13 @Pennwick it now adds up the temperatures of the contained beakers when mixing

**Comment by Deantwo** [comment]
I'd point out that you can't heat a bottle. But whatever.
I still don't know which chemicals this is actually useful for.

**Pull request merged**